### DISCUSSION TITLE
Downloading Raw Full Network Captures with PCAP and Stenographer (for all events)

### DISCUSSION THREAD

**Discussion by GIF-POP** [Security-Onion-Solutions/securityonion#13400] [Unsupported Versions]
Hi there. I am new to Security Onion, and I have been requested to provide a raw full network captures on the network segment that the Security Onion sensor is monitoring. This capture is to be similar to what you can get with Wireshare or other network packet analyzers. I went through the Security Onion essentials videos, read through the Security Onion 2.4 documentation, and reviewed Stenographer utility information on Github.

After performing some trials, it looks like I can only obtain network packets on an event basis. For example, ICMP, DNS, DHCP, specific malware identifier(s), etc. need to be entered as part of the query. I cannot seem to just ask for a complete dump of the network segment or even refine the query for a full network capture dump on all network traffic types concerning the endpoints in question.

Again, being a newbie to Security Onion, I most likely missed something obvious, so any suggestions would be appreciated.

Thanks. --George

**Comment by dougburks** [comment]
You should be able to specify a query for a network segment. Here's an example from the Stenographer documentation at https://github.com/google/stenographer#querying:
```
net 1.0.0.0/8
```

For more information about doing large Stenographer exports in Security Onion, please see:
https://docs.securityonion.net/en/2.4/stenographer.html#command-line

You may also want to consider switching from Stenographer to Suricata for your PCAP retention as Suricata stores full packet capture in standard PCAP files that can be read by other libpcap tools. For more information, please see:
https://docs.securityonion.net/en/2.4/suricata.html#pcap
